api/src/services/initial-replication.js

```diff
@@ -0,0 +1,39 @@
+const db = require('../db');
+const authorization = require('./authorization');
+const purgedDocs = require('./purged-docs');
+const _ = require('lodash');
+const { DOC_IDS_WARN_LIMIT } = require('../services/replication-limit-log');
+
+const getContext = async (userCtx) => {
+  const info = await db.medic.info();
+  const authContext = await authorization.getAuthorizationContext(userCtx);
+  const docsByReplicationKey = await authorization.getDocsByReplicationKey(authContext);
+
+  const excludeTombstones = { includeTombstones: false };
+  const allowedIds = authorization.filterAllowedDocIds(authContext, docsByReplicationKey, excludeTombstones);
+  const unpurgedIds = await purgedDocs.getUnPurgedIds(userCtx.roles, allowedIds);
+
+  const excludeTombstonesAndTasks = { includeTombstones: false, includeTasks: false };
+  const warnIds = authorization.filterAllowedDocIds(authContext, docsByReplicationKey, excludeTombstonesAndTasks);
+  const unpurgedWarnIds = _.intersection(unpurgedIds, warnIds);
+
+  return {
+    docIds: unpurgedIds,
+    warnDocIds: unpurgedWarnIds,
+    warn: unpurgedWarnIds.length >= DOC_IDS_WARN_LIMIT,
+    limit: DOC_IDS_WARN_LIMIT,
+    lastSeq: info.update_seq,
+  };
+};
+
+const getDocIdsRevPairs = async (docIds) => {
+  const result = await db.medic.allDocs({ keys: docIds });
+  return result.rows
+    .filter(row => row.value && row.value.rev)
```

Comment:
Maybe also filter out deleted docs? Or will they already be filtered out of the `docIds`?

Reply:
Deleted docs would be excluded automatically, since the ids are view query results. The only way a deleted doc can make it here is if it is deleted between the view request and this one.